script.js

```diff
@@ -0,0 +1,136 @@
+        deleteNoteBtn.appendChild(deleteButtonText); 
+
+        //adding the buttons to their container 
+        buttonsDiv.appendChild(deleteNoteBtn); 
+        buttonsDiv.appendChild(editNoteBtn);
+
+        // add buttons container to the li 
+        listItem.appendChild(buttonsDiv); 
+
+        
+        // add event listener to the delete button
+        deleteNoteBtn.addEventListener("click",function(){
+            let confirmationResult = confirm("Are you sure you want to delete this note?"); 
+            if(confirmationResult){
+                notes.splice(index,1); 
+                showNotes();
+            }
+        });
+        // add event listener to the edit button
+        editNoteBtn.addEventListener("click",function(){
+            // add classes to change the border and background according to the button state (edit/confirm edit)
+            this.classList.toggle("confirm");
+            this.classList.toggle("edit");
+            // determine the text inside the button 
+            if(this.textContent ==="Edit"){
+                this.textContent = "Confirm";
+            }else{
+                this.textContent = "Edit"; 
+            }
+
+            if(this.classList.contains("confirm")){
+                const form = createForm(note.title,note.description,index);
+                listItem.removeChild(divTitle);
+                listItem.removeChild(divDescription); 
+                listItem.insertBefore(form,listItem.lastChild);
+            }else{
+                const editedTitle = document.querySelector(".edit-form .title"+index).value; 
+                const editedDescription = document.querySelector(".edit-form .description"+index).value;
+                notes.splice(index,1,{title:editedTitle,description:editedDescription}); 
+                showNotes();
+            }
+        });
+        list.appendChild(listItem);
+    });
+    localStorage.setItem("notes",JSON.stringify(notes));
+}
+// create the form that will appear when editing
+function createForm(titleContent,descriptionContent,index){
+    const form = document.createElement("form");
+    form.classList.add("edit-form"); 
+
+    const editTitle = document.createElement("input");
+    editTitle.classList.add("title");
+    editTitle.classList.add("title"+index);
+    editTitle.value = titleContent; 
+
+    const editDescription = document.createElement("textarea"); 
+    editDescription.classList.add("description"+index);
+    editDescription.rows = "8"; 
+    editDescription.cols = "60";
+    editDescription.value = descriptionContent;
+
+    form.appendChild(editTitle); 
+    form.appendChild(editDescription); 
+    
+    return form; 
+}
+
+
+// function to sort notes array elements depending on the title 
+function sortElements(){
+    notes.sort(function(a,b){
+        let compareValue = 0;
+        a.title<b.title? compareValue = -1 : compareValue = 1; 
+        return compareValue;
+    });
```

Comment:
You can add more functionality to the sort method by converting all letters to uppercase or lowercase in order to ignore if the letters are uppercase or lowercase.

Reply:
I'll do it.